### PR TITLE
Use caplog in FoVBackgroundMaker tests

### PR DESCRIPTION
**Pull request by Jaleleddine**
**Description**

This pull request adds caplog in test_fov

**Dear reviewer**

### CODE REVIEW

File: gammapy/makers/background/tests/test_fov.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+from typing import List, Dict
```
Comment (adonath) [inline gammapy/makers/background/tests/test_fov.py:2]: It seems the import is unused...please remove.

**Comment by adonath** [comment]
Thanks a lot @Jaleleddine, it seems there is still a test fail: https://github.com/gammapy/gammapy/pull/3203/checks?check_run_id=1763067685#step:9:1876, please fix. 

If I understood correctly `caplog` is a way to test logging output. In general we should probably add an entry in the developer documentation for this: https://docs.gammapy.org/0.18.2/development/howto.html, so that people know how to use it for other tests.

Otherwise the PR looks good to me!

**Review by registerrier — COMMENTED** [review]
Thanks @Jaleleddine .

I think the CI fail comes from the absence of warning raised in the `FoVBackgroundMaker` with the `"fit"` option. This was recently removed and I think is a good indication that such tests are needed.

One connected issue is #2216  . At the time, the decision on the technical solution was delayed before deciding on a good choice for the logging itself. 
It seems that log testing is nevertheless needed and that `logging` is probably the solution we are going to keep at least for some time. 

Is the practical solution here satisfactory? I think that we should systematically test that "WARNING" are present in the 'log'. Should we test the log message as well?

**Comment by registerrier** [comment]
BTW, any idea why test fails on linux and not on the MacOS CI?

**Comment by adonath** [comment]
@registerrier I fully agree, that we will just keep in using `logging` for now and I agree it's good to have more systematic testing for this. However in general I would say this is already very detailed work and I think for now we do the testing on a best effort basis, where we think important warnings are emitted...

@registerrier Is the CI fail because the failing fit is emitted as info and not a warning? https://github.com/gammapy/gammapy/blob/master/gammapy/makers/background/fov.py#L145? I have no idea why this just appears on Linux...maybe @Bultako knows?

**Comment by adonath** [comment]
@Jaleleddine I think you accidentally pushed to the wrong branch: https://github.com/gammapy/gammapy/tree/caplog, can you maybe check and push to this PR instead? Let me know (Slack) if you need help with that...

File: gammapy/makers/background/tests/test_fov.py [hunk 2]
```diff
@@ -131,7 +126,7 @@ def test_fov_bkg_maker_fit_nocountsnobackground(obs_dataset, exclusion_mask, cap
 
     assert caplog.records[-1].levelname == "WARNING"
     assert "Fit did not converge for test-fov" in caplog.records[-1].message
-    assert "Background model parameters might be unphysical for test-fov" in caplog.records[-1].message
+    #assert "Background model parameters might be unphysical for test-fov" in caplog.records[-1].message
```
Comment (registerrier) [inline gammapy/makers/background/tests/test_fov.py:129]: Please remove commented line

**Review by registerrier — APPROVED** [review]
Thanks @Jaleleddine . See inline comment

**Review by adonath — APPROVED** [review]
Thanks a lot @Jaleleddine, I'll proceed and merge this PR now, please add an entry in the developer documentation in a follow up PR.

**Pull request merged**